### PR TITLE
Added Swagger details for CheckNameAvailability API

### PR DESCRIPTION
**Pull request by amchandn**
Old PR link(closed): https://github.com/Azure/azure-rest-api-specs/pull/4900/files/dcf9fbef154d7c3c2321f7bbc65b15d1a137f5fe

I've incorporated the suggested changes and replied to all the comments in old PR

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5617 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5617/commits/aba9ab69171c49363c2bbfdd57f887214147f3e3

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json [hunk 1]
```diff
@@ -18,6 +18,64 @@
     "application/json"
   ],
   "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/locations/{location}/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "RecoveryServices"
+        ],
+        "summary": "API to check for resource name availability.\r\nA name is available if no other resource exists that has the same SubscriptionId, Resource Name and Type\r\nor if one or more such resources exist, each of these must be GCed and their time of deletion be more than 24 Hours Ago",
```
Comment (nschonni) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json]:
```suggestion
        "summary": "API to check for resource name availability.\r\nA name is available if no other resource exists that has the same SubscriptionId, Resource Name and Type\r\nor if one or more such resources exist, each of these must be GC'd and their time of deletion be more than 24 Hours Ago",
```

Comment (amchandn) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json]: Done. Resolving the comment

**Comment by dsgouda** [comment]
@jhendrixMSFT Looks like you reviewed the older PR, would be faster if you took over this

**Comment by jhendrixMSFT** [comment]
@amchandn is this new API deployed?

**Comment by amchandn** [comment]
Yes, @jhendrixMSFT this new API is deployed.

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
ARM approval is already received for the same changes (in a different PR) https://github.com/Azure/azure-rest-api-specs/pull/4900/

**Comment by dsgouda** [comment]
@amchandn Please join the Azure github org [here](https://aka.ms/azuregithub)

**Comment by amchandn** [comment]
@dsgouda , Yes, I'm already a member of Azure github org

**Pull request merged**